### ISSUE TITLE
CKV_AWS_107: AWS::IAM::Policy for action ecr:GetAuthorizationToken with resource "*" for CloudFormation

### ISSUE THREAD

**Issue by adamjq** [bridgecrewio/checkov#1815]
**Describe the bug**

Checkov returns an error `Check: CKV_AWS_107: "Ensure IAM policies does not allow credentials exposure"` when using a CloudFormation IAM Policy with action `ecr:GetAuthorizationToken` and `resource "*"`. `ecr:GetAuthorizationToken` doesn't allow for ARN restriction so the rule should not apply to this action. AWS CDK also generates this policy by default when using ECR with an ECS Fargate service.

A previous PR https://github.com/bridgecrewio/checkov/pull/1032 introduced this fix for Terraform but not CloudFormation.

It also appears that a change was introduced in checkov version `2.0.486` as previous versions don't produce errors.

**To Reproduce**

1. Create a directory `cfn` and add this IAM policy `iam.yaml`:
```cloudformation
AWSTemplateFormatVersion: "2010-09-09"
Description: IAM policy
Resources:
  ECRGetAuthTokenPolicy:
    Type: 'AWS::IAM::Policy'
    Properties:
      PolicyName: ECRGetAuthTokenPolicy
      PolicyDocument:
        Statement:
          - Action: 'ecr:GetAuthorizationToken'
            Effect: Allow
            Resource: '*'
```

2. Run `docker run -t -v ${PWD}:/cfn bridgecrew/checkov:2.0.505 -d cfn` to see failure at latest version `2.0.505`

3. Run `docker run -t -v ${PWD}:/cfn bridgecrew/checkov:2.0.486 -d cfn` to see failure at version `2.0.486`

4. Run `docker run -t -v ${PWD}:/cfn bridgecrew/checkov:2.0.486 -d cfn` to see passing at version `2.0.485`

**Expected behavior**

`AWS::IAM::Policy` with Action `'ecr:GetAuthorizationToken'` and Resource `'*'` should pass checkov checks.

**Desktop (please complete the following information):**
 - OS: Mac OS 11.6
 - Checkov Version 2.0.505

**Additional context**

This issue was previously raised as a feature request here https://github.com/bridgecrewio/checkov/issues/988

A pull request to fix the issue for Terraform was previously merged here https://github.com/bridgecrewio/checkov/pull/1032
